### PR TITLE
[clojure] Fix @field highlighting bug

### PR DESCRIPTION
**Pull request by NoahTheDuke**
Fixes error introduced in #5307.

Add tests for Clojure highlighting.

### CODE REVIEW

File: queries/clojure/highlights.scm [hunk 1]
```diff
@@ -103,7 +103,7 @@
 ((sym_lit) @method
  (#lua-match? @method "^[.][^-]"))
 ((sym_lit) @field
- (#lua-match? @field "^[.]-"))
+ (#lua-match? @field "^\\.-"))
```
Comment (lucario387) [inline queries/clojure/highlights.scm:106]: What about %. instead of \\., since it's lua-match

Comment (NoahTheDuke) [reply queries/clojure/highlights.scm:106]: Sure, I'll try that out.

**Comment by lucario387** [comment]
```
Fail	||	highlight queries tests/query/highlights/clojure/test.clj	
            ...eesitter/nvim-treesitter/tests/query/highlights_spec.lua:96: Error in at tests/query/highlights/clojure/test.clj:10:1: expected "keyword", captures: { "symbol" }", highlights: { "symbol" }
            Expected objects to be the same.
            Passed in:
            (nil)
            Expected:
            (boolean) true
            
            stack traceback:
            	...eesitter/nvim-treesitter/tests/query/highlights_spec.lua:96: in function 'check_assertions'
            	...eesitter/nvim-treesitter/tests/query/highlights_spec.lua:119: in function <...eesitter/nvim-treesitter/tests/query/highlights_spec.lua:118>
```
Seems like something's wrong here

**Comment by NoahTheDuke** [comment]
Indeed. It passes locally so I'm unsure how to approach a solution. Any advice?

**Comment by lucario387** [comment]
> Indeed. It passes locally so I'm unsure how to approach a solution. Any advice?

I'm not sure how it passed locally on your end. Mine doesn't pass as well

Having `:Inspect` while the cursor at `clojure/test:clj:10:1` yields this

```
Treesitter                                                                                                                                                                                      
  - @symbol.clojure links to @symbol clojure    
```

**Comment by lucario387** [comment]
The node that fails, has its name being `kwd_lit`, so not changed in this PR. You probably applied the wrong test for it

**Comment by NoahTheDuke** [comment]
Turns out, for some reason, my custom `after` queries were being applied, so that's my fault. Hopefully I've gotten it right locally so the tests will now pass.

File: queries/clojure/highlights.scm [hunk 2]
```diff
@@ -100,11 +100,17 @@
...
 
 ; Interop
-((sym_lit) @method
- (#lua-match? @method "^[.][^-]"))
-((sym_lit) @field
- (#lua-match? @field "^%.-"))
-((sym_lit) @field
+(list_lit
+ .
+ (sym_lit) @method
```
Comment (lucario387) [inline queries/clojure/highlights.scm:105]: Can I get some docs on how these methods/fields in clojure work?

And why is it necessary to limit it to the only the first `sym_lit` of each list

File: tests/query/highlights/clojure/test.clj
Comment (lucario387) [inline tests/query/highlights/clojure/test.clj:1]: Please don't add these many tests, only nodes that were relevant to this PR. No need to test for punctuations, brackets and such, so is include, string, variable ...

Comment (NoahTheDuke) [reply tests/query/highlights/clojure/test.clj:1]: Sure, I'll trim it down.

File: queries/clojure/highlights.scm (hunk 2 above)
Comment (NoahTheDuke) [reply queries/clojure/highlights.scm:105]: I'll remove this change, put it in a different PR.

File: tests/query/highlights/clojure/test.clj
Comment (NoahTheDuke) [reply tests/query/highlights/clojure/test.clj:1]: If you don't mind, why remove those tests?

Comment (lucario387) [reply tests/query/highlights/clojure/test.clj:1]: Couple of reasons
- Highlight tests here should be used to reflect the specific differences between nvim-treesitter and the upstream parser creator. If there's no differences then there's no need for those tests as it's already tested from upstream.
- Tests can also be used to detect regressions, so tests for some changes in cases like this is nice, but only for those regressions

**Review by lucario387 — APPROVED** [review]

**Review by ObserverOfTime — APPROVED** [review]

**Pull request merged**